test/ecto/adapters/postgres_test.exs

```diff
@@ -707,6 +707,35 @@ defmodule Ecto.Adapters.PostgresTest do
     assert all(query) == String.trim(result)
   end
 
+  test "explain_query" do
+    assert_raise(RuntimeError, "version not supported", fn ->
+      SQL.explain_query("SELECT 1", "9.4")
+    end)
+
+    assert_raise(RuntimeError, "version not supported", fn ->
+      SQL.explain_query("SELECT 1", 0)
+    end)
+
+    assert_raise(ArgumentError, "bad boolean value T", fn ->
+      SQL.explain_query("SELECT 1", "12.0", explain: "T")
+    end)
+
+    assert SQL.explain_query("SELECT 1", "9.5") ==
```

Comment:
Calling `explain` with an empty `opts` could resolve into a `EXPLAIN {query}` (no options) but I decided to include all of them with their default values to be more explicit, which may help when analyzing logs and debugging. Wdyt?